tests/Operation/ListSearchIndexesTest.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace MongoDB\Tests\Operation;
+
+use MongoDB\Exception\InvalidArgumentException;
+use MongoDB\Operation\ListSearchIndexes;
+
+class ListSearchIndexesTest extends TestCase
+{
+    /** @dataProvider provideInvalidStringValues */
+    public function testConstructorExpectNameFilterToBeAString($name): void
```

Comment:
If you end up changing how `name` is passed, I suppose you could fold this into `testConstructorOptionTypeChecks()`.